### PR TITLE
Initial client auth for XDS

### PR DESCRIPTION
**Pull request by costinm**
The auth was lost when we removed the Envoy sidecar in front of Istiod. 

Uses the same authenticator as the CA.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[x] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[x] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by costinm** [comment]
Issue/24008

**Comment by costinm** [comment]
> Any reason to use this approach rather than changing
> 
> https://github.com/istio/istio/blob/65bd2ac632cbd6ec8e3fd85bb9874f685aea0cfd/pilot/pkg/bootstrap/server.go#L591
> 
> ?

We want to allow auth with JWT as well - and keep things consistent with the cert path. 

And in future we'll want to actually get the identity from the cert - and check the namespace.

File: pilot/pkg/proxy/envoy/v2/ads.go [hunk 1]
```diff
@@ -194,12 +196,42 @@ func receiveThread(con *XdsConnection, reqChannel chan *xdsapi.DiscoveryRequest,
...
+	peerInfo, ok := peer.FromContext(ctx)
 	peerAddr := "0.0.0.0"
 	if ok {
 		peerAddr = peerInfo.Addr.String()
 	}
 
+	// Authenticate - currently just checks that request has a certificate signed with the our key.
+	// Protected by flag to avoid breaking upgrades - should be enabled in multi-cluster/meshexpansion where
+	// XDS is exposed.
+	if s.Authenticators != nil && len(s.Authenticators) > 0 {
```
Comment (howardjohn) [inline pilot/pkg/proxy/envoy/v2/ads.go:209]: nit: break this out into a function, this method is already massive

File: pilot/pkg/proxy/envoy/v2/ads.go [hunk 2]
```diff
@@ -194,12 +196,42 @@ func receiveThread(con *XdsConnection, reqChannel chan *xdsapi.DiscoveryRequest,
...
+			adsLog.Infoa("Authenticated XDS: ", peerInfo, " ", authenticatedID.AuthSource, " ", authenticatedID.Identities)
+		} else {
+			// TODO: add a flag to prevent unauthenticated requests ( 15010 )
+			// request not over TLS ( on the insecure port
+			adsLog.Infoa("Unauthenticated XDS: ", peerInfo)
+		}
+
+	}
+
+	// TODO: We should validate that the namespace in the cert matches the claimed namespace in metadata.
```
Comment (howardjohn) [inline pilot/pkg/proxy/envoy/v2/ads.go:233]: nit: put todo inside auth

**Comment by linsun** [comment]
@costinm thanks for fixing this!  Can you get the tests passing?

File: pilot/pkg/proxy/envoy/v2/ads.go (hunk 2 above)
Comment (costinm) [reply pilot/pkg/proxy/envoy/v2/ads.go]: I moved it where the implementation should be - next to the code getting the metadata.

**Pull request merged**